indexer/src/utils/filter.test.ts

```diff
@@ -155,3 +154,55 @@ Deno.test("isKakarotTransaction: `to` address matching KAKAROT_ADDRESS", () => {
   const success = isKakarotTransaction(transaction);
   assertEquals(success, true);
 });
+
+Deno.test(
+  "isRevertedWithOutOfResources: true on status reverted and revert reason",
+  () => {
+    const receipt: TransactionReceipt = {
+      executionStatus: "EXECUTION_STATUS_REVERTED",
+      transactionHash: "0x01",
+      transactionIndex: "0x01",
+      actualFee: "0x01",
+      contractAddress: "0x01",
+      l2ToL1Messages: [],
+      events: [],
+      revertReason: "RunResources has no remaining steps",
+    };
+    const success = isRevertedWithOutOfResources(receipt);
+    assertEquals(success, true);
+  },
+);
+
+Deno.test("isRevertedWithOutOfResources: false on status succeeded", () => {
+  const receipt: TransactionReceipt = {
```

Comment:
note: so the TransactionReceipt coming from apibara allows for success status with revertReason, which is a bit surprising

Reply:
I think this is a random test to verify that we index the run of resources transactions if and only if the following conditions are met:

```
receipt.executionStatus.includes("REVERTED") &&
    receipt.revertReason?.includes("RunResources has no remaining steps")
```

And if the revert reason is the right message but the execution status is not reverted, then we don't index. But I'm not sure this case is possible in reality.

Reply:
Don't think it can but better to test everything.

Reply:
if you are able to build an object of type `TransactionReceipt` with these two conditions, it's that the type itself doesn't exclude them mutually. A basic example of a type enforcing this is

```
interface BaseResponse {
  // Add other shared fields here
  id: number;
  timestamp: Date;
}

type SuccessResponse = BaseResponse & {
  a: "SUCCESS";
  b?: undefined;
};

type FailureResponse = BaseResponse & {
  a: "FAILURE";
  b: string;
};

type Response = SuccessResponse | FailureResponse;
```

```
const successResponse: Response = {
  id: 1,
  timestamp: new Date(),
  a: "SUCCESS"
}; // valid

const failureResponse: Response = {
  id: 2,
  timestamp: new Date(),
  a: "FAILURE",
  b: "Error message"
}; // valid

const invalidResponse1: Response = {
  id: 3,
  timestamp: new Date(),
  a: "SUCCESS",
  b: "Should be undefined"
}; // Type error

const invalidResponse2: Response = {
  id: 4,
  timestamp: new Date(),
  a: "FAILURE"
}; // Type error
```

courtesy of chat gpt